wolfcrypt/src/sp_int.c

```diff
@@ -178,20 +183,23 @@ int sp_unsigned_bin_size(sp_int* a)
  * a     SP integer.
  * in    Array of bytes.
  * inSz  Number of data bytes in array.
- * returns BAD_FUNC_ARG when the number is too big to fit in an SP and
+ * returns MP_VAL when the number is too big to fit in an SP and
            MP_OKAY otherwise.
  */
-int sp_read_unsigned_bin(sp_int* a, const byte* in, int inSz)
+int sp_read_unsigned_bin(sp_int* a, const byte* in, word32 inSz)
 {
     int err = MP_OKAY;
     int i, j = 0, k;
 
     /* Extra digit added to SP_INT_DIGITS to be used in calculations. */
-    if (inSz > (SP_INT_DIGITS - 1) * (int)sizeof(a->dp[0])) {
+    if (inSz > SP_INT_DIGITS * (int)sizeof(a->dp[0])) {
```

Comment:
The `SP_INT_DIGITS - 1` change was made here https://github.com/wolfSSL/wolfssl/pull/3097 after the DO178C branch. I think the original master code is newer and the correct version.

Reply:
@dgarske, the cert version seems correct to me.
@SparkiDev, thoughts?

Reply:
SP_INT_DIGITS - 1 is the correct change.
Should only fill to maximum number of supported bits so that operations can use the extra digit in calculations.

Reply:
Thanks, @dgarske and @SparkiDev!, Update code with suggested changes.